frontend/src/Editor/Components/FilePicker.jsx

```diff
@@ -95,6 +96,47 @@ export const FilePicker = ({ width, height, component, currentState, onComponent
   const [showSelectdFiles, setShowSelectedFiles] = React.useState(false);
   const [selectedFiles, setSelectedFiles] = React.useState([]);
 
+  /**
+   * *getFileData()
+   * @param {*} file
+   * @param {*} method: readAsDataURL, readAsText
+   */
+  const getFileData = (file = {}, method = 'readAsText') => {
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = (result) => {
+        resolve([result, reader]);
+      };
+      reader[method](file);
+      reader.onerror = (error) => {
+        reject(error);
+        if (error.name == 'NotReadableError') {
+          toast.error(error.message, { hideProgressBar: true, autoClose: 3000 });
+        }
+      };
+    }).then((result) => {
+      if (method === 'readAsDataURL') {
+        return result[0].srcElement.result.split(',')[1];
+      }
+      return result[0].srcElement.result;
+    });
+  };
+
+  const fileReader = async (file) => {
+    // * readAsText
+    const readFileAsText = await getFileData(file);
+
+    // * readAsDataURL
+    const readFileAsDataURL = await getFileData(file, 'readAsDataURL');
+
+    return {
+      name: file.name,
+      type: file.type,
+      content: readFileAsText,
+      data: readFileAsDataURL,
```

Comment:
I'm thinking about whether it will be better to expose this field as dataUrl instead of data itself such that we don't split result when doing using `readAsDataURL` method. The reasoning being users can use it directly without any preprocessing. Wdyt?

Reply:
yeah, dataURL makes sense